models/safe/zkevm/safe_zkevm_transactions.sql

```diff
@@ -0,0 +1,63 @@
+{{
+    config(
+        materialized='incremental',
+
+        alias= 'transactions',
+        partition_by = ['block_month'],
+        unique_key = ['block_date', 'tx_hash', 'trace_address'],
+        file_format ='delta',
+        incremental_strategy='merge',
+        post_hook='{{ expose_spells(\'["zkevm"]\',
+                                    "project",
+                                    "safe",
+                                    \'["danielpartida"]\') }}'
+    )
+}}
+
+select
+    'zkevm' as blockchain,
+    try_cast(date_trunc('day', tr.block_time) as date) as block_date,
+    CAST(date_trunc('month', tr.block_time) as DATE) as block_month,
+    tr.block_time,
+    tr.block_number,
+    tr.tx_hash,
+    s.address,
+    tr.to,
+    tr.value,
+    tr.gas,
+    tr.gas_used as execution_gas_used,
+    et.gas_used as total_gas_used,
+    tr.tx_index,
+    tr.sub_traces,
+    tr.trace_address,
+    tr.success,
+    tr.error,
+    tr.code,
+    tr.input,
+    tr.output,
+    case
+        when bytearray_substring(tr.input, 1, 4) = 0x6a761202 then 'execTransaction'
+        when bytearray_substring(tr.input, 1, 4) = 0x468721a7 then 'execTransactionFromModule'
+        when bytearray_substring(tr.input, 1, 4) = 0x5229073f then 'execTransactionFromModuleReturnData'
+        else 'unknown'
+    end as method
+from {{ source('zkevm', 'traces') }} tr
+join {{ ref('safe_zkevm_safes') }} s
+    on s.address = tr."from"
+join {{ ref('safe_zkevm_singletons') }} ss
+    on tr.to = ss.address
+join {{ source('zkevm', 'transactions') }} et
```

Comment:
add incremental filters on `et` alias in join conditions

Reply:
Can you make a suggestion? I am not really sure what you mean here

Reply:
Maybe something like
```
{% if is_incremental() %}
and {{ incremental_predicate('tr.blocktime') }}
{% endif %}
```
after `on`.